constellations/orion/src/repos/account_repo.go

```diff
@@ -117,32 +118,82 @@ func (acc *accountRepo) SelectAllNegativeBalances() ([]domains.AccountSum, error
 	return results, nil
 }
 
-func (acc *accountRepo) Insert(account domains.Account) error {
+func (acc *accountRepo) InsertWithUser(account domains.Account, user domains.User) error {
+	tx, err := acc.db.Begin()
+	if err != nil {
+		return err
+	}
 	utils.LogWithContext("accountRepo.Insert", logger.Fields{"account": account})
 	statement := "INSERT INTO accounts (" +
 		"created_at, " +
 		"updated_at, " +
 		"primary_email, " +
 		"password" +
 		") VALUES (?, ?, ?, ?)"
-
 	stmt, err := acc.db.Prepare(statement)
 	if err != nil {
+		tx.Rollback()
 		return appErrors.WrapDbPrepare(err, statement)
 	}
 	defer stmt.Close()
 
 	now := time.Now().UTC()
-	execResult, err := stmt.Exec(
+	_, err = stmt.Exec(
 		now,
 		now,
 		account.PrimaryEmail,
 		account.Password,
 	)
 	if err != nil {
-		return appErrors.WrapDbExec(err, statement, account)
+		tx.Rollback()
+		return err
+	}
+
+	statement2 := "INSERT INTO users (" +
+		"created_at, " +
+		"updated_at, " +
+		"first_name, " +
+		"last_name," +
+		"middle_name, " +
+		"email," +
+		"phone, " +
+		"is_guardian," +
+		"account_id," +
+		"notes," +
+		"school," +
+		"graduation_year" +
+		") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+	stmt2, err := acc.db.Prepare(statement2)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+	defer stmt2.Close()
+
+	_, err = stmt2.Exec(
+		now,
+		now,
+		user.FirstName,
+		user.LastName,
+		user.MiddleName,
+		user.Email,
+		user.Phone,
+		user.IsGuardian,
+		user.AccountId,
+		user.Notes,
+		user.School,
+		user.GraduationYear,
+	)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
same thing. We can return the err. However, if you look in the appErrors package, you'll see we can wrap this around a `WrapDbTxCommit.

If the transaction failed to commit for some reason, I think we also need to rollback.
so
```
if err != nil {
    tx.Rollback()
    return appErrors.WrapDbCommit(err, ...)
}
```